opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/SimpleQueryString.java

```diff
@@ -0,0 +1,114 @@
+      (b, v) -> b.fuzzyTranspositions(Boolean.parseBoolean(v.stringValue()));
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> lenient =
+      (b, v) -> b.lenient(Boolean.parseBoolean(v.stringValue()));
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> minimumShouldMatch =
+      (b, v) -> b.minimumShouldMatch(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> quoteFieldSuffix =
+      (b, v) -> b.quoteFieldSuffix(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> boost =
+      (b, v) -> b.boost(Float.parseFloat(v.stringValue()));
+
+  ImmutableMap<Object, Object> argAction = ImmutableMap.builder()
+      .put("analyze_wildcard", analyzeWildcard)
+      .put("analyzer", analyzer)
+      .put("auto_generate_synonyms_phrase_query", autoGenerateSynonymsPhraseQuery)
+      .put("flags", flags)
+      .put("fuzzy_max_expansions", fuzzyMaxExpansions)
+      .put("fuzzy_prefix_length", fuzzyPrefixLength)
+      .put("fuzzy_transpositions", fuzzyTranspositions)
+      .put("lenient", lenient)
+      .put("default_operator", defaultOperator)
+      .put("minimum_should_match", minimumShouldMatch)
+      .put("quote_field_suffix", quoteFieldSuffix)
+      .put("boost", boost)
+      .build();
+
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    Iterator<Expression> iterator = func.getArguments().iterator();
+    var fields = (NamedArgumentExpression) iterator.next();
+    var query = (NamedArgumentExpression) iterator.next();
+    SimpleQueryStringBuilder queryBuilder = QueryBuilders
+      .simpleQueryStringQuery(query.getValue().valueOf(null).stringValue())
+      .fields(parseFields(fields.getValue().valueOf(null).collectionValue()));
+    while (iterator.hasNext()) {
+      NamedArgumentExpression arg = (NamedArgumentExpression) iterator.next();
+      if (!argAction.containsKey(arg.getArgName())) {
+        throw new SemanticCheckException(String
+            .format("Parameter %s is invalid for simple_query_string function.", arg.getArgName()));
+      }
+      ((BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder>) argAction
+          .get(arg.getArgName()))
+          .apply(queryBuilder, arg.getValue().valueOf(null));
+    }
+    return queryBuilder;
+  }
+
+  private Map<String, Float> parseFields(List<ExprValue> input) {
+    try {
+      var res = new HashMap<String, Float>();
+      for (var literal : input) {
+        var elem = literal.stringValue();
+        if (!elem.contains("^")) {
+          res.put(elem, 1F);
+          continue;
+        }
+
+        var parts = elem.split("\\^");
+        var weight = Float.parseFloat(parts[parts.length - 1]);
+        var field = Arrays.stream(parts).limit(parts.length - 1).collect(Collectors.joining("^"));
```

Comment:
I'd like to look at making field weight a first-class parser concept. Not clear if this can differentiate between field named `title^4.3` and weight of 4.3 for field `title`.

Is the `\\^` above to allow for escaping `^` that is part of field names?

Reply:
I proposed the following in the opensearch-project -- https://github.com/opensearch-project/sql/issues/182#issuecomment-1130404678

Reply:
> Is the `\\^` above to allow for escaping `^` that is part of field names?

No. `split` function receives a `regex`, so I have to escape `^`, otherwise it would be interpreted as a regex control symbol.

A user can make a search like
> `simple_query_string(["some_field^1", "my^favourite^field^ever^100500"], ...)`

and it would be parsed into following map
```
/-------------------------v--------\
|          field          | weight |
>-------------------------+--------<
| some_field              | 1      |
| my^favourite^field^ever | 100500 |
\-------------------------^--------/
```